piff/stats.py

```diff
@@ -0,0 +1,264 @@
+        :param bin_size:    Logarithmic size of separation bins.
+        :param sky_coordinates: Are we in sky coordinates or otherwise? [default: False]
+        :param sep_units:   I think only matters if in sky_coordinates.
+                            Can be 'rad', 'arcmin', 'deg', etc.
+                            Sets the units of separation in min_sep, max_sep
+        :param ra_units:    I think only matters if in sky_coordinates.
+                            Can be 'rad', 'arcmin', 'deg', etc.
+                            Says what the units of the ra coordinate is.
+        :param dec_units:   I think only matters if in sky_coordinates.
+                            Can be 'rad', 'arcmin', 'deg', etc.
+                            Says what the units of the dec coordinate is.
+
+        note: assumes coord[:,0] = ra, coord[:,1] = dec or x/y
+
+        :returns: logr, rho1, rho2, rho3, rho4, rho5 correlation functions
+
+
+        Notes
+        -----
+        From Jarvis:2015 p 10, eqs 3-18 - 3-22.
+        e = e_psf ; de = e_psf - e_model
+        T is size
+
+        rho1 = < de* de >
+        rho2 = < e* de >  (in the rowe paper this is < e* de + de* e >
+        rho3 = < (e* dT / T) (e dT / T) >
+        rho4 = < de* (e dT / T) >
+        rho5 = < e* (e dT / T) >
+
+        We calculate these quantities using treecorr
+
+        also note that for gN = gNr + i gNi = gN1 + i gN2:
+        xi.xip[k] += g1rg2r + g1ig2i;       // g1 * conj(g2)
+        xi.xip_im[k] += g1ig2r - g1rg2i;
+        xi.xim[k] += g1rg2r - g1ig2i;       // g1 * g2
+        xi.xim_im[k] += g1ig2r + g1rg2i;
+
+        so since we probably really want, e.g. 0.5 < e* de + de* e >, then we
+        can just use xip.
+
+        """
+        from .gaussian_model import Gaussian
+        import treecorr
+
+        hsm = Gaussian()
+
+        # measure moments with Gaussian on image
+        if logger:
+            logger.info("Measuring Stars")
+        shapes_truth = np.array([ hsm.fitStar(star).getParameters() for star in stars ])
+        # from stars get positions
+        if logger:
+            logger.info("Getting Star Positions")
+        positions = np.array([ self.interp.getStarPosition(star) for star in stars ])
+        # generate the model stars
+        if logger:
+            logger.info("Generating Model Stars")
+        # stars_model = []
+        # for position in positions:
+        #     print(position)
+        #     stars_model.append(self.generate_star(position))
+        stars_model = np.array([ self.generate_star(position) for position in positions ])
+        # measure moments with Gaussian on interpolated model image
+        if logger:
+            logger.info("Measuring Model Stars")
+        shapes_model = np.array([ hsm.fitStar(star).getParameters() for star in stars_model ])
+
+        # define terms for the catalogs
+        ra = positions[:, 0]
+        dec = positions[:, 1]
```

Comment:
I don't think these are actually ra, dec.  At least in what we have so far, `getStarPosition` returns the position in field coordinates, not celestial coordinates.  So these should be u,v.